projects/kit/components/carousel/carousel.component.ts

```diff
@@ -130,11 +130,12 @@ export class TuiCarouselComponent {
         return index < this.index || index > this.index + this.itemsCount;
     }
 
-    protected onIntersection(
-        {intersectionRatio}: IntersectionObserverEntry,
-        index: number,
-    ): void {
-        if (intersectionRatio && intersectionRatio !== 1 && !this.transitioned) {
+    protected onIntersection(entry?: IntersectionObserverEntry, index = 0): void {
```

Comment:
Is it even possible to not have no entries here and still have a callback? Sounds like we are overcomplicating things here for no reason.

Reply:
fixed